elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java

```diff
@@ -62,7 +65,8 @@ public RequestScope(JsonApiDocument jsonApiDocument,
                         JsonApiMapper mapper,
                         AuditLogger auditLogger,
                         MultivaluedMap<String, String> queryParams,
-                        SecurityMode securityMode) {
+                        SecurityMode securityMode,
+                        Class<? extends PermissionExecutor> permissionExecutorClass) {
```

Comment:
It holds caching and state that is relative to your request; it cannot be easily made global.